pageserver/src/tenant_mgr.rs

```diff
@@ -77,27 +78,34 @@ pub fn init_tenant_mgr(
                         );
                     }
                 } else {
+                    // This case happens if we crash during attach before creating the attach marker file
+                    let is_empty = tenant_dir_path.is_empty_dir().with_context(|| {
+                        format!("Failed to check whether {tenant_dir_path:?} is an empty dir")
+                    })?;
+                    if is_empty {
+                        info!("removing empty tenant directory {tenant_dir_path:?}");
+                        if let Err(e) = std::fs::remove_dir(&tenant_dir_path) {
+                            error!(
+                                "Failed to remove empty tenant directory '{}': {e:#}",
+                                tenant_dir_path.display()
+                            )
+                        }
+                        continue;
+                    }
+
+                    let tenant_ignore_mark_file = tenant_dir_path.join(IGNORED_TENANT_FILE_NAME);
+                    if tenant_ignore_mark_file.exists() {
+                        info!("Found an ignore mark file {tenant_ignore_mark_file:?}, skipping the tenant");
+                        continue;
```

Comment:
What if we call attach for that tenant?

Reply:
I suppose you get tenant that is `Attaching` for a brief period of time, then transitions into `Broken` state after error

```
calling Tenant::attach with a tenant directory that doesn't have the attaching marker file:\ntenant_dir: {}\nmarker_file: {}"
```

Reply:
Good point, I should have added a negative test on attaches/loads, since I've even missed a few points here.

Reworked now to ensure both code paths treat first local files: with this approach, `attach` disallows local files, which simplifies a few things.
Still, not fond how both startup and load logic have to treat all these files differently.